Buildings/Controls/OBC/CDL/Logical/Validation/VariablePulseNegativeStartTime.mo

```diff
@@ -0,0 +1,117 @@
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Pulse conPul(
+    final amplitude=0.65,
+    final width=0.8,
+    final period=3600)
+    "Continuous pulse signal"
+    annotation (Placement(transformation(extent={{20,50},{40,70}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant con(
+    final k=0.75)
+    "Constant"
+    annotation (Placement(transformation(extent={{-80,-70},{-60,-50}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant con1(
+    final k=0)
+    "Constant value"
+    annotation (Placement(transformation(extent={{-80,50},{-60,70}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Pulse conPul1(
+    final amplitude=0.65,
+    final width=0.85,
+    final period=3600,
+    final shift=360)
+    "Continuous pulse signal"
+    annotation (Placement(transformation(extent={{20,-10},{40,10}})));
+  Buildings.Controls.OBC.CDL.Logical.VariablePulse tru(
+    final samplePeriod=120,
+    final period=900)
+    "Variable pulse with the constant width input"
+    annotation (Placement(transformation(extent={{-40,-10},{-20,10}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant con2(
+    final k=1)
+    "Constant value"
+    annotation (Placement(transformation(extent={{-80,-10},{-60,10}})));
+  Buildings.Controls.OBC.CDL.Logical.VariablePulse falTru1(
+    final samplePeriod=120,
+    final period=900)
+    "Variable pulse with width input changing from zero to positive value"
+    annotation (Placement(transformation(extent={{60,-70},{80,-50}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Pulse conPul2(
+    final amplitude=0.5,
+    final width=0.5,
+    final period=3600,
+    final shift=300,
+    final offset=0.3)    "Continuous pulse signal"
+    annotation (Placement(transformation(extent={{20,-70},{40,-50}})));
+equation
+  connect(con.y, conTru.uWid)
+    annotation (Line(points={{-58,-60},{-42,-60}}, color={0,0,127}));
+  connect(con1.y, fal.uWid)
+    annotation (Line(points={{-58,60},{-42,60}},color={0,0,127}));
+  connect(conPul.y, truFal.uWid)
+    annotation (Line(points={{42,60},{58,60}},    color={0,0,127}));
+  connect(conPul1.y, falTru.uWid)
+    annotation (Line(points={{42,0},{58,0}},      color={0,0,127}));
+  connect(con2.y, tru.uWid)
+    annotation (Line(points={{-58,0},{-42,0}},  color={0,0,127}));
+  connect(conPul2.y, falTru1.uWid)
+    annotation (Line(points={{42,-60},{58,-60}}, color={0,0,127}));
+annotation (experiment(
+      StartTime=-3600,
+      StopTime=3600,
+      Tolerance=1e-06),
+  __Dymola_Commands(
+      file="modelica://Buildings/Resources/Scripts/Dymola/Controls/OBC/CDL/Logical/Validation/VariablePulseNegativeStartTime.mos" "Simulate and plot"),
+  Documentation(info="<html>
+<p>
+Validation test for the block
+<a href=\"modelica://Buildings.Controls.OBC.CDL.Logical.VariablePulse\">
+Buildings.Controls.OBC.CDL.Logical.VariablePulse</a>.
+</p>
+<p>
+This validates the blocks with a start time of <i>-3600</i>.
```

Comment:
Add units, and better write <i>t=-3600</i> seconds.

Reply:
updated.